pysr/regressor_sequence.py

```diff
@@ -0,0 +1,425 @@
+
+        assert recursive_history_length is not None and recursive_history_length > 0
+
+        # Try to load model from <equation_file>.pkl
+        print(f"Checking if {pkl_filename} exists...")
+        if os.path.exists(pkl_filename):
+            print(f"Loading model from {pkl_filename}")
+            assert binary_operators is None
+            assert unary_operators is None
+            assert n_features_in is None
+            model = cls(*pysr_args, **pysr_kwargs)
+            with open(pkl_filename, "rb") as f:
+                model._regressor = pkl.load(f)
+            # Change equation_file_ to be in the same dir as the pickle file
+            base_dir = os.path.dirname(pkl_filename)
+            base_equation_file = os.path.basename(model._regressor.equation_file_)
+            model._regressor.equation_file_ = os.path.join(base_dir, base_equation_file)
+
+            # Update any parameters if necessary, such as
+            # extra_sympy_mappings:
+            model._regressor.set_params(**pysr_kwargs)
+            if "equations_" not in model.__dict__ or model.equations_ is None:
+                model._regressor.refresh()
+
+            model.recursive_history_length = recursive_history_length
+            return model
+
+        # Else, we re-create it.
+        print(
+            f"{pkl_filename} does not exist, "
+            "so we must create the model from scratch."
+        )
+        assert binary_operators is not None or unary_operators is not None
+        assert n_features_in is not None
+        assert recursive_history_length is not None
+
+        # TODO: copy .bkup file if exists.
+        model = cls(
+            equation_file=str(equation_file),
+            binary_operators=binary_operators,
+            unary_operators=unary_operators,
+            **pysr_kwargs,
+        )
+
+        model._regressor.nout_ = nout
+        model._regressor.n_features_in_ = n_features_in
+
+        if feature_names_in is None:
+            model._regressor.feature_names_in_ = np.array(
+                [f"x{i}" for i in range(n_features_in)]
+            )
+            model._regressor.display_feature_names_in_ = np.array(
+                [f"x{_subscriptify(i)}" for i in range(n_features_in)]
+            )
+        else:
+            assert len(feature_names_in) == n_features_in
+            model._regressor.feature_names_in_ = feature_names_in
+            model._regressor.display_feature_names_in_ = feature_names_in
+
+        if selection_mask is None:
+            model._regressor.selection_mask_ = np.ones(n_features_in, dtype=np.bool_)
+        else:
+            model._regressor.selection_mask_ = selection_mask
+
+        model._regressor.refresh(checkpoint_file=equation_file)
+        model.recursive_history_length = recursive_history_length
+
+        return model
+
+    def __repr__(self):
+        return self._regressor.__repr__().replace(
+            "PySRRegressor", "PySRSequenceRegressor"
+        )
+
```

Comment:
Is there a cleaner way to do this? `replace` replaces **all** occurrences so this might cause some issues if PySRRegressor appears as a variable or shows up somewhere else in the string for whatever reason

Reply:
since __repr__ in PySRRegressor starts with "PySRRegressor.equations_ = [" I changed it to only replace the first occurrence :)